torchgeo/datamodules/nasa_marine_debris.py

```diff
@@ -3,35 +3,14 @@
 
 """NASA Marine Debris datamodule."""
 
-from typing import Any, Dict, List, Optional
-
-import matplotlib.pyplot as plt
-import pytorch_lightning as pl
-import torch
-from torch import Tensor
-from torch.utils.data import DataLoader
+from typing import Any
 
 from ..datasets import NASAMarineDebris
+from .geo import NonGeoDataModule
 from .utils import dataset_split
 
 
-def collate_fn(batch: List[Dict[str, Tensor]]) -> Dict[str, Any]:
-    """Custom object detection collate fn to handle variable boxes.
-
-    Args:
-        batch: list of sample dicts return by dataset
-
-    Returns:
-        batch dict output
-    """
-    output: Dict[str, Any] = {}
-    output["image"] = torch.stack([sample["image"] for sample in batch])
-    output["boxes"] = [sample["boxes"] for sample in batch]
```

Comment:
@calebrob6 starting a thread here so we can discuss how to handle NASA Marine Debris.

I think the solution in this `collate_fn` is actually correct, we want a list of tensors, not a single tensor: https://github.com/kornia/kornia/pull/1497

Note that the name is wrong, we should be used bbox_xyxy or bbox_xywh instead of bbox (boxes is translated to bbox internally in our `AugmentationSequential` wrapper) but we can fix that when working on https://github.com/microsoft/torchgeo/discussions/985.

Reply:
Okay, see if the last commit fixes this.